crates/vm/src/arch/vm.rs

```diff
@@ -592,15 +521,26 @@ where
         })
     }
 
-    /// Same as [`VmExecutor::create_initial_state`] but sets more information for performance
-    /// metrics when feature "perf-metrics" is enabled.
+    /// Same as [`InterpretedInstance::create_initial_state`] but sets more information for
+    /// performance metrics when feature "perf-metrics" is enabled.
```

Comment:
not a fan of this duplication. i feel it should be a separate function that takes config (and maybe an optional reference to pk or that can be a separate method) as input and not a struct method.

Reply:
let me move this to a constructor of `VmState` in https://github.com/openvm-org/openvm/pull/1926